views.sql

```diff
@@ -0,0 +1,143 @@
+CREATE VIEW webinar_financial_report AS
+SELECT Webinar.id, CAST(SUM(Webinar.price) / 100.0 AS DECIMAL(10, 2)) AS income
+FROM Webinar
+         INNER JOIN StudentWebinar on Webinar.id = StudentWebinar.webinar_id
+GROUP BY Webinar.id;
+GO
+
+CREATE VIEW course_financial_report AS
+SELECT Course.id,
+       CAST((dbo.course_full_income(Course.id) + dbo.course_advance_income(Course.id)) /
+            100.0 AS DECIMAL(10, 2)) AS income
+FROM Course;
+GO
+
+CREATE VIEW studies_financial_report AS
+SELECT Studies.id,
+       CAST(((dbo.studies_registration_income(Studies.id) + SUM(dbo.semester_income(Semester.id))) /
+             100.0) AS DECIMAL(10, 2)) AS income
+FROM Studies
+         INNER JOIN Semester ON Studies.id = Semester.studies_id
+GROUP BY Studies.id;
+GO
+
+CREATE VIEW students_who_purchased_meeting AS
+SELECT Student.id, Student.first_name, Student.last_name, COUNT(*) AS purchased_meetings
+FROM StudentMeeting
+         INNER JOIN Student ON StudentMeeting.student_id = Student.id
+GROUP BY Student.id, Student.first_name, Student.last_name;
+GO
+
+CREATE VIEW debtor_list AS
+SELECT DISTINCT TOP 100 PERCENT Student.id, Student.first_name, Student.last_name
```

Comment:
why 100? why not 87? or 21?
Consider pagination
https://learn.microsoft.com/en-us/sql/t-sql/queries/select-order-by-clause-transact-sql?view=sql-server-ver16#using-offset-and-fetch-to-limit-the-rows-returned